### PR TITLE
Adding tooltips for dataintegration and embedding

### PR DESCRIPTION
**Pull request by seb-mueller**
# Background

Adding some tooltips. I've struggled to get the placement right, maybe reviewing engeneer could give some pointers onto howdo better place them.

#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-705
#### Link to staging deployment URL 

https://ui-seb-ui231-v2.scp-staging.biomage.net/
#### Anything else the reviewers should know about the changes here

https://docs.google.com/presentation/d/1DsLbRBeYHaHYQbURVukF1UiueR60imemynSLqLgXEAI/edit?skip_itp2_check=true#slide=id.gc6021a98ad_0_31

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [x] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR
image

### CODE REVIEW

**Comment by xverges** [comment]
@seb-mueller I see that at least one of the tooltip has a link: we can make it actionable. Take a look at https://github.com/biomage-ltd/ui/blob/90d42db2ef2c6f4518f418a622560bc9cdeaf58e/src/components/data-processing/Classifier/CalculationConfig.jsx#L21 to see how they can be more than plain text. Can work with you on it tomorrow if you want
image

File: src/components/data-processing/ConfigureEmbedding/CalculationConfig.jsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ import {
 } from '../../../redux/actions/experimentSettings';
 
 import updateCellSetsClustering from '../../../redux/actions/cellSets/updateCellSetsClustering';
+import { types } from 'mime-types';
```
Comment (cosa65) [inline src/components/data-processing/ConfigureEmbedding/CalculationConfig.jsx]: I don't see this is being used anywhere so it should probably be removed.

File: src/components/data-processing/ConfigureEmbedding/CalculationConfig.jsx [hunk 2]
```diff
@@ -195,8 +206,19 @@ const CalculationConfig = (props) => {
...
+          The parameter is, in a sense, a guess about the number of close neighbors each cell has.
+          In most implementations, perplexity defaults to 30. This focuses the attention of t-SNE on preserving the
+          distances to its 30 nearest neighbors and puts virtually no weight on preserving distances to the remaining points.
+          The perplexity value has a complex effect on the resulting pictures.'>
+            <QuestionCircleOutlined />
+          </Tooltip>
         </Form.Item>
+
         <Form.Item label='Learning rate'>
+
```
Comment (cosa65) [inline src/components/data-processing/ConfigureEmbedding/CalculationConfig.jsx]: Nitpick: could you remove these 3 empty spaces?

File: src/components/data-processing/DataIntegration/SeuratV4Options.jsx [hunk 3]
```diff
@@ -40,6 +45,15 @@ const SeuratV4Options = (props) => {
             },
           })}
         />
+        {/* Identifies features that are outliers on a 'mean variability plot'.
```
Comment (cosa65) [inline src/components/data-processing/DataIntegration/SeuratV4Options.jsx]: Are these comments intended?

File: src/pages/experiments/[experimentId]/data-processing/index.jsx [hunk 4]
```diff
@@ -279,7 +279,8 @@ const DataProcessingPage = ({ experimentId, experimentData, route }) => {
     {
       key: 'configureEmbedding',
       name: 'Configure embedding',
-      description: 'The number of dimensions used to configure the embedding is set here. This dictates the number of clusters in the Uniform Manifold Approximation and Projection (UMAP) which is taken forward to the ‘Data Exploration’ page.',
+      // description: 'The number of dimensions used to configure the embedding is set here. This dictates the number of clusters in the Uniform Manifold Approximation and Projection (UMAP) which is taken forward to the ‘Data Exploration’ page.',
```
Comment (cosa65) [inline src/pages/experiments/[experimentId]/data-processing/index.jsx]: Remove this commented line.

File: src/components/data-processing/ConfigureEmbedding/CalculationConfig.jsx (hunk 2 above)
Comment (seb-mueller) [reply src/components/data-processing/ConfigureEmbedding/CalculationConfig.jsx]: done

**Comment by vickymorrison** [comment]
Generally, the tooltip text looks good. A few comments:
1. the links should be clickable.
2. the positioning of the ? icons is a little off in places. If possible, it's best to be consistent with the placement of others such as in the filters i.e. after the text and before the menu/box. E.g. in data integration it should be "#HGV genes: ? 2000"

Some specific comments about the text:
3. In the 'exclude gene categories' tooltip is it possible to provide a link to the lists of genes that are included in each category?
4. In 'Distance metric' change "Euclidean is has been used for most data-sets so far." to something like "Euclidean is the standard for most normalized datasets" or similar.
5. In the 'Resolution' tooltip this is contradictory "recovers smaller, and therefore larger number...." and should be re-worded.

**Review by cosa65 — APPROVED** [review]
lgtm

**Comment by seb-mueller** [comment]
So all requested changes should now been fixed, thanks everyone who helped me with this one!

One outstanding point:
3. In the 'exclude gene categories' tooltip is it possible to provide a link to the lists of genes that are included in each category?
 This is a good idea, but since this is not been implemented yet, we can't say at this point. We can revisit once this functionality is implemented!

**Comment by vickymorrison** [comment]
Placement of one of the ? is still off (see screenshot). But otherwise good to go.
Screenshot 2021-04-30 at 19 50 24

**Pull request merged**